config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -169,6 +209,43 @@ private Optional<ConfigDocument> getLatestVersionConfigDoc(
     return Optional.empty();
   }
 
+  private Map<ConfigResourceContext, Optional<ConfigDocument>> getLatestVersionConfigDocs(
+      Set<ConfigResourceContext> configResourceContexts) throws IOException {
+    // build filter
+    Filter[] childFilters = new Filter[configResourceContexts.size()];
+    int index = 0;
+    for (ConfigResourceContext configResourceContext : configResourceContexts) {
+      childFilters[index] = getConfigResourceContextFilter(configResourceContext);
+      index += 1;
+    }
+    Filter filter = new Filter();
+    filter.setOp(Filter.Op.OR);
+    filter.setChildFilters(childFilters);
+
+    // build query
+    Query query = new Query();
+    query.setFilter(filter);
+    query.addOrderBy(new OrderBy(VERSION_FIELD_NAME, false));
+    query.setLimit(configResourceContexts.size());
+
+    Map<ConfigResourceContext, Optional<ConfigDocument>> latestVersionConfigDocs = new HashMap<>();
```

Comment:
Similar about order. The main point is that if a user passes in a set of update requests, they should be processed and returned in that order. The existing test that was forced to change should be a flashing red indicator that was broken.